frontend/public/components/utils/tile-view-page.jsx

```diff
@@ -495,12 +494,46 @@ export class TileViewPage extends React.Component {
   }
 
   componentDidMount() {
-    const { items, filterGroups, getAvailableFilters, groupByTypes } = this.props;
+    const {
+      items,
+      filterGroups,
+      getAvailableFilters,
+      groupByTypes,
+      storeFilterKey,
+      filterRetentionPreference,
+      defaultActiveFilters,
+    } = this.props;
     const { categories } = this.state;
     const availableFilters = getAvailableFilters(defaultFilters, items, filterGroups);
-
+    if (!(storeFilterKey && localStorage.getItem(storeFilterKey)) && defaultActiveFilters) {
+      _.each(Object.keys(defaultActiveFilters), (filterGroup) => {
+        _.each(defaultActiveFilters[filterGroup], (key) => {
+          availableFilters[filterGroup][key].active = true;
+        });
+      });
+    }
     const activeValues = getActiveValuesFromURL(availableFilters, filterGroups, groupByTypes);
 
+    if (storeFilterKey && localStorage.getItem(storeFilterKey)) {
+      try {
+        const lastFilters = JSON.parse(localStorage.getItem(storeFilterKey));
+        if (lastFilters && activeValues.activeFilters) {
+          filterRetentionPreference &&
+            _.each(filterRetentionPreference, (filterGroup) => {
+              lastFilters[filterGroup] &&
+                _.each(Object.keys(lastFilters[filterGroup]), (key) => {
+                  if (lastFilters[filterGroup][key].active) {
+                    activeValues.activeFilters[filterGroup][key].active = true;
+                  }
+                });
+            });
+        }
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Failed while parsing -> ', e);
```

Comment:
```suggestion
        console.error('Failed parsing user filter settings.');
```
We don't really need debug logic in production code... we can log an error (instead of swallowing it) just so if it doesn't work on production we can see it in the logs. Why doesn't really matter, and if it does Chrome does support live editing of code on production for us to log it out.

Reply:
updated